src/sql/workbench/contrib/executionPlan/browser/topOperationsTab.ts

```diff
@@ -0,0 +1,406 @@
+		stack.push(...graph.root.children);
+		while (stack.length !== 0) {
+			const node = stack.pop();
+			const row: { [key: string]: any } = {};
+			node.topOperationsData.forEach((d, i) => {
+				let displayText = d.displayValue.toString();
+				if (i === 0) {
+					row[d.columnName] = {
+						displayText: displayText,
+						linkOrCommand: 'placeholder',
+						dataType: d.dataType
+					};
+				} else {
+					row[d.columnName] = {
+						text: displayText,
+						ariaLabel: d.displayValue,
+						dataType: d.dataType
+					};
+				}
+				if (columnValues.indexOf(d.columnName) === -1) {
+					columnValues.splice(i, 0, d.columnName);
+				}
+			});
+			row['nodeId'] = node.id;
+			if (node.children) {
+				node.children.forEach(c => stack.push(c));
+			}
+
+			dataMap.push(row);
+		}
+		const columns = columnValues.map((c, i) => {
+			return <Slick.Column<any>>{
+				id: c.toString(),
+				name: c,
+				field: c.toString(),
+				formatter: i === 0 ? hyperLinkFormatter : textFormatter,
+				sortable: true,
+			};
+		});
+
+		const topOperationContainer = DOM.$('.top-operations-container');
+		this._container.appendChild(topOperationContainer);
+		const header = this._instantiationService.createInstance(ExecutionPlanViewHeader, topOperationContainer, {
+			planIndex: index,
+		});
+		header.query = graph.query;
+		header.relativeCost = graph.root.relativeCost;
+		const tableContainer = DOM.$('.table-container');
+		topOperationContainer.appendChild(tableContainer);
+		this._topOperationsContainers.push(topOperationContainer);
+
+		const rowNumberColumn = new RowNumberColumn({ numberOfRows: dataMap.length });
+		columns.unshift(rowNumberColumn.getColumnDefinition());
+
+		let copyHandler = new CopyKeybind<any>();
+		copyHandler.onCopy(e => {
+
+			const selectedDataRange = selectionModel.getSelectedRanges()[0];
+			let csvString = '';
+			if (selectedDataRange) {
+				const data = [];
+
+				for (let rowIndex = selectedDataRange.fromRow; rowIndex <= selectedDataRange.toRow; rowIndex++) {
+					const dataRow = table.getData().getItem(rowIndex);
+					const row = [];
+					for (let colIndex = selectedDataRange.fromCell; colIndex <= selectedDataRange.toCell; colIndex++) {
+						const dataItem = dataRow[table.columns[colIndex].field];
+						if (dataItem) {
+							row.push(dataItem.displayText ?? dataItem.text);
+						} else {
+							row.push(' ');
+						}
+					}
+					data.push(row);
+				}
+				csvString = data.map(row =>
+					row.map(x => `${x}`).join('\t')
+				).join('\n');
+
+				const columns = [];
+
+				for (let colIndex = selectedDataRange.fromCell; colIndex <= selectedDataRange.toCell; colIndex++) {
+					columns.push(table.columns[colIndex].name);
+				}
+
+			}
+
+			this._instantiationService.createInstance(CopyTableData).run({
+				selectedText: csvString
+			});
+		});
+
+		const selectionModel = new CellSelectionModel<any>();
+
+		const table = new Table<any>(tableContainer, {
```

Comment:
Please give this an actual type - at the very least it should be Slick.SlickData if you don't have a more specific type you can give it